adapters/kosovo.js

```diff
@@ -0,0 +1,209 @@
+  return row;
+}
+
+function getStationMeasurements(rawStation, parameters) {
+  const location = rawStation.shift();
+  const measurement = {
+    location : location,
+    city : staticData[location].city,
+    date : getDate(rawStation.shift()),
+    coordinates : staticData[location].coordinates,
+    attribution : attribution,
+    averagingPeriod : averagingPeriod
+  };
+
+  const stationMeasurements = [];
+  rawStation.forEach((rawMeasurement, index) => {
+    if (rawMeasurement)
+        stationMeasurements.push(Object.assign({
+          parameter : parameters[index].parameter,
+          unit : parameters[index].unit,
+          value : Number(rawMeasurement.replace(',', '.'))
+        }, measurement));
+  });
+  return stationMeasurements;
+}
+
+function getMeasurements(rawStations, parameters) {
+  const measurements = [];
+  rawStations.forEach((rawStation) =>
+    measurements.push(...getStationMeasurements(rawStation, parameters))
+  );
+  return measurements;
+}
+
+function getTable(rawData) {
+  const trs = rawData.children.filter(child => child.name === 'tr');
+  log.debug('--------------- raw rows -------------');
+  log.debug(trs);
+  const rawHeaders = trs.shift();
+  let headers = getRow(rawHeaders.children);
+  headers.splice(0, 2); // Remove station and date header columns.
+  log.debug('--------------- headers -------------');
+  log.debug(headers);
+  const parameters = getParameters(headers);
+  log.debug('--------------- parameters -------------');
+  log.debug(parameters);
+  const rawStations = trs.map(tr => getRow(tr.children));
+  log.debug('--------------- raw stations -------------');
+  log.debug(rawStations);
+
+  const measurements = getMeasurements(rawStations, parameters)
+  log.debug('--------------- stations -------------');
+  log.debug(measurements);
+  return measurements;
+}
+
+async function getKosovoAQ(source) {
+  return new Promise(function(resolve, reject) {
+    getKosovoAQHTML(source.url)
+    .then(html => {
+      log.debug('--------------- html -------------');
+      log.debug(html);
+      const rawJSON = getKosovoAQlatestRawJSON(html);
+      log.debug('--------------- raw JSON -------------');
+      log.debug(rawJSON);
+      // jo2: Probably should use JSONata or something similar to do a query for the root data object.
+      // As long as the source HTML page structure is not modified, this should work.
+      const rawTable = rawJSON._root.children[2].children[2].children[4].children[1].children[5].children[1].children[1].children[1];
+
+      log.debug('--------------- raw JSON table -------------');
+      log.debug(rawTable);
+      const table = getTable(rawTable).filter(measurement => openaq_parameters.includes(measurement.parameter));
+      resolve({ name : module.exports.name, measurements : table });
+    })
+    .catch(error => reject(error));
+  });
+}
+
+module.exports.name = 'kosovo';
```

Comment:
Can you please move this up top to right below the imports to keep it consistent with other adapters?